app/component.js

```diff
@@ -0,0 +1,130 @@
+import LayoutManager from 'layoutmanager';
+import $ from 'jquery';
+import _ from 'underscore';
+
+// Cache registered components into this object.
+const components = {};
+
+class ComponentError extends Error {
+  constructor(message) {
+    super({ message });
+
+    this.message = message;
+  }
+}
+
+/**
+ * Represents a Component
+ *
+ * @class
+ */
+class Component extends LayoutManager {
+  constructor() {
+    super();
+
+    // Not sure why super doesn't call the correct constructor :-/
+    LayoutManager.prototype.constructor.apply(this, arguments);
+
+    // Attach the dataset.
+    this.dataset = this.$el.data();
+
+    // Activate new components found after rendering.
+    this.listenTo(this, 'afterRender', function() {
```

Comment:
why `this.listenTo(this)`? `this.on` will just create a self-reference, so it's not a memleak concern, right?